src/DockAreaTitleBar.cpp

```diff
@@ -149,10 +149,11 @@ class CTitleBarButton : public tTitleBarButton
 		if(QEvent::EnabledChange == ev->type() && HideWhenDisabled)
 		{
 			// force setVisible() call 
-			setVisible(isEnabled());
+			//setVisible(isEnabled()); // Calling setVisible() directly here doesn't work well when button is expected to be shown first time
```

Comment:
I would remove the part that says `setVisible(isEnabled());` and just leave the explanation

Reply:
done